From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Sat, 13 Jun 2020 17:01:48 +1200
Subject: backport: Add ConfigurationService shim to BalloonService

---
 webapp/src/Service/BalloonService.php | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/webapp/src/Service/BalloonService.php b/webapp/src/Service/BalloonService.php
index 81c6aec07..512beaf4a 100644
--- a/webapp/src/Service/BalloonService.php
+++ b/webapp/src/Service/BalloonService.php
@@ -10,6 +10,14 @@ use Doctrine\ORM\EntityManagerInterface;
 
 class BalloonService
 {
+    // backport: use a shim for the ConfigurationService class (added in DOMjudge 7.3)
+    protected $config;
+    /** @required */
+    public function setConfig(\App\Service\ConfigurationService $config)
+    {
+        $this->config = $config;
+    }
+
     /**
      * @var EntityManagerInterface
      */
-- 
2.20.1

