From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Wed, 8 Apr 2020 01:33:39 +1200
Subject: Add commands nzpc-install and nzpc-bare-install to dj_setup_database

---
 sql/dj_setup_database.in | 20 +++++++++++++++++---
 1 file changed, 17 insertions(+), 3 deletions(-)

diff --git a/sql/dj_setup_database.in b/sql/dj_setup_database.in
index 584d2dbf7..cce7633b3 100755
--- a/sql/dj_setup_database.in
+++ b/sql/dj_setup_database.in
@@ -33,6 +33,10 @@ Commands:
   install-examples install examples only
   upgrade          upgrade MySQL database schema to current version
 
+NZPC Commands:
+  nzpc-install      create database with example contest, load NZPC configuration
+  nzpc-bare-install create database, load NZPC configuration
+
 Options:
   -u <user>  connect to MySQL with DB admin <user>
   -p <pass>  use password <pass> for DB admin user
@@ -283,17 +287,27 @@ create-db-users)
 	verbose "Created empty database and users."
 	;;
 
-bare-install|install)
+bare-install|install|nzpc-bare-install|nzpc-install)
 	read_dbpasswords
 	create_db_users
 	export DB_FIRST_INSTALL=1
 	symfony_console doctrine:migrations:migrate -n
 	DBUSER=$domjudge_DBUSER PASSWD=$domjudge_PASSWD symfony_console -vv domjudge:db-config:update
 	unset DB_FIRST_INSTALL
-	if [ "$1" = "install" ]; then
+	if [ "$1" = "nzpc-bare-install" ] || [ "$1" = "nzpc-install" ]; then
+		# load the NZPC configuration
+		DBUSER=$domjudge_DBUSER PASSWD=$domjudge_PASSWD symfony_console -vv nzpc:configure --all
+	fi
+	if [ "$1" = "install" ] || [ "$1" = "nzpc-install" ]; then
+		# load the example data
 		DBUSER=$domjudge_DBUSER PASSWD=$domjudge_PASSWD symfony_console domjudge:load-example-data
+		if [ "$1" = "nzpc-install" ]; then
+			# load NZPC-specific example data and enable registration
+			DBUSER=$domjudge_DBUSER PASSWD=$domjudge_PASSWD symfony_console doctrine:fixtures:load --group=NZPCFixtureFixes --append
+			DBUSER=$domjudge_DBUSER PASSWD=$domjudge_PASSWD symfony_console -vv nzpc:configure --enable-registration
+		fi
 	fi
-	if [ "$1" = "install" ]; then
+	if [ "$1" = "install" ] || [ "$1" = "nzpc-install" ]; then
 		verbose "SQL structure and default/example data installed."
 	else
 		verbose "SQL structure and defaults installed (no sample data)."
-- 
2.20.1

