From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Fri, 31 Jul 2020 08:22:54 +1200
Subject: C# currently works in chroot, remove old comments about errors

---
 misc-tools/dj_make_chroot.in | 2 --
 1 file changed, 2 deletions(-)

diff --git a/misc-tools/dj_make_chroot.in b/misc-tools/dj_make_chroot.in
index 65b1fcd50..6a7586249 100755
--- a/misc-tools/dj_make_chroot.in
+++ b/misc-tools/dj_make_chroot.in
@@ -129,7 +129,6 @@ INCLUDEDEBS="ca-certificates"
 # Packages to install after upgrade (space separated):
 INSTALLDEBS="default-jre-headless pypy python3 locales"
 # For C# support add: mono-runtime
-# However running mono within chroot still gives errors...
 
 # Packages to remove after upgrade (space separated):
 REMOVEDEBS=""
@@ -158,7 +157,6 @@ INCLUDEDEBS="software-properties-common"
 # Packages to install after upgrade (space separated):
 INSTALLDEBS="default-jre-headless pypy python3 locales"
 # For C# support add: mono-runtime
-# However running mono within chroot still gives errors...
 
 # Packages to remove after upgrade (space separated):
 REMOVEDEBS=""
-- 
2.20.1

