From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Thu, 26 Mar 2020 04:07:34 +1300
Subject: Modify domserver Dockerfile to build from source instead of release
 tarball

---
 docker/domserver/Dockerfile | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/docker/domserver/Dockerfile b/docker/domserver/Dockerfile
index 2dc461ea2..77e38785c 100644
--- a/docker/domserver/Dockerfile
+++ b/docker/domserver/Dockerfile
@@ -32,8 +32,11 @@ RUN php composer-setup.php \
     && mv /composer.phar /usr/local/bin/composer
 
 # Add DOMjudge source code and build script
-ADD domjudge.tar.gz /domjudge-src
-ADD domserver/build.sh /domjudge-src
+COPY . /domjudge-src/domjudge
+ADD docker/domserver/build.sh /domjudge-src
+
+# Bootstrap the configure/build system
+RUN make -C /domjudge-src/domjudge dist
 
 # Build and install domserver
 RUN /domjudge-src/build.sh
@@ -70,7 +73,7 @@ RUN php composer-setup.php \
 # Copy domserver directory and add script files
 COPY --from=domserver-build /opt/domjudge/domserver /opt/domjudge/domserver
 COPY --from=domserver-build /opt/domjudge/doc /opt/domjudge/doc
-COPY ["domserver/scripts", "/scripts/"]
+COPY ["docker/domserver/scripts", "/scripts/"]
 
 # Add user, create PHP FPM socket dir, change permissions for domjudge directory and fix scripts
 RUN useradd -m domjudge \
@@ -85,7 +88,7 @@ RUN useradd -m domjudge \
 CMD ["/scripts/start.sh"]
 
 # Copy supervisor files
-COPY ["domserver/supervisor", "/etc/supervisor/conf.d/"]
+COPY ["docker/domserver/supervisor", "/etc/supervisor/conf.d/"]
 
 # Expose HTTP port
 EXPOSE 80
-- 
2.20.1

