From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Thu, 23 Jul 2020 01:01:22 +1200
Subject: Don't allocate a TTY when building the judgehost docker image

---
 docker/build.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/docker/build.sh b/docker/build.sh
index e8b64d0b8..ce6454119 100755
--- a/docker/build.sh
+++ b/docker/build.sh
@@ -28,7 +28,7 @@ echo "[ok] Done building Docker image for domserver"
 echo "[..] Building Docker image for judgehost using intermediate build image..."
 docker build -t domjudge/judgehost:${VERSION}-build -f judgehost/Dockerfile.build .
 docker rm -f domjudge-judgehost-${VERSION}-build > /dev/null 2>&1 || true
-docker run -it --name domjudge-judgehost-${VERSION}-build --privileged domjudge/judgehost:${VERSION}-build
+docker run --name domjudge-judgehost-${VERSION}-build --privileged domjudge/judgehost:${VERSION}-build
 docker cp domjudge-judgehost-${VERSION}-build:/chroot.tar.gz .
 docker cp domjudge-judgehost-${VERSION}-build:/judgehost.tar.gz .
 docker rm -f domjudge-judgehost-${VERSION}-build
-- 
2.20.1

