From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Tue, 21 Jul 2020 05:48:15 +1200
Subject: Bootstrap from source instead of downloading release tarball in
 docker build script

---
 docker/build.sh | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/docker/build.sh b/docker/build.sh
index 14c07432e..2e9910e30 100755
--- a/docker/build.sh
+++ b/docker/build.sh
@@ -15,6 +15,7 @@ fi
 
 VERSION="$1"
 
+if false; then
 URL=https://www.domjudge.org/releases/domjudge-${VERSION}.tar.gz
 FILE=domjudge.tar.gz
 
@@ -27,6 +28,14 @@ then
 fi
 
 echo "[ok] DOMjudge version ${VERSION} downloaded as domjudge.tar.gz"; echo
+fi
+
+echo "[..] Boostrapping..."
+tar c -C .. --exclude=./.git --exclude="./docker/*.tar.gz" . |
+	docker build --target=dist -t domjudge/dist:${VERSION} -f docker/domserver/Dockerfile.source -
+docker run --rm domjudge/dist:${VERSION} tar cz -C /domjudge-src domjudge > domjudge.tar.gz
+docker rmi --no-prune domjudge/dist:${VERSION}
+echo "[ok] Done boostrapping"
 
 echo "[..] Building Docker image for domserver..."
 ./build-domjudge.sh "domjudge/domserver:${VERSION}"
-- 
2.20.1

