From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Tue, 21 Jul 2020 05:48:15 +1200
Subject: Bootstrap from source instead of downloading release tarball in
 docker build script

---
 docker/build.sh | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/docker/build.sh b/docker/build.sh
index 6cbd189b1..929eebb03 100755
--- a/docker/build.sh
+++ b/docker/build.sh
@@ -47,6 +47,7 @@ URL=https://www.domjudge.org/releases/domjudge-${VERSION}.tar.gz
 FILE=domjudge.tar.gz
 section_end
 
+if false; then
 section_start "Download DOMjudge tarball"
 echo "[..] Downloading DOMjudge version ${VERSION}..."
 if ! wget --quiet "${URL}" -O ${FILE}
@@ -56,6 +57,14 @@ then
 fi
 echo "[ok] DOMjudge version ${VERSION} downloaded as domjudge.tar.gz"; echo
 section_end
+fi
+
+echo "[..] Boostrapping..."
+tar c -C .. --exclude=./.git --exclude="./docker/*.tar.gz" . |
+	docker build --target=dist -t "${NAMESPACE}/dist:${VERSION}" -f docker/domserver/Dockerfile.source -
+docker run --rm "${NAMESPACE}/dist:${VERSION}" tar cz -C /domjudge-src domjudge > domjudge.tar.gz
+docker rmi --no-prune "${NAMESPACE}/dist:${VERSION}"
+echo "[ok] Done boostrapping"
 
 section_start "Build domserver container"
 echo "[..] Building Docker image for domserver..."
-- 
2.20.1

