From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Tue, 28 Jul 2020 06:20:32 +1200
Subject: Install C# compiler in judgehost image

---
 docker/judgehost/Dockerfile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/docker/judgehost/Dockerfile b/docker/judgehost/Dockerfile
index 1fcb59624..d776abb59 100644
--- a/docker/judgehost/Dockerfile
+++ b/docker/judgehost/Dockerfile
@@ -18,6 +18,7 @@ RUN apt-get update \
 	php-cli php-zip php-gd php-curl php-mysql php-json \
 	php-gmp php-xml php-mbstring python3 \
 	gcc g++ default-jre-headless default-jdk-headless ghc fp-compiler \
+	mono-mcs \
 	&& rm -rf /var/lib/apt/lists/*
 
 # Set up user
-- 
2.20.1

